app/lib/common/pages/medications/cubit.dart

```diff
@@ -13,18 +13,43 @@ class MedicationsCubit extends Cubit<MedicationsState> {
   final int _id;
 
   Future<void> loadMedications() async {
-    final requestUri = annotationServerUrl.replace(
-      path: 'api/v1/medications/$_id',
-    );
     emit(MedicationsState.loading());
-    final response = await http.get(requestUri);
-    if (response.statusCode != 200) {
+    final response = await sendRequest();
+    if (response == null){
       emit(MedicationsState.error());
-      return;
     }
+    if (response == null) return;
```

Comment:
Put the return statement in the same if body as above

Reply:
ah sorry that was stupid